lib/Window.js

```diff
@@ -149,6 +149,7 @@ module.exports = (Item, registry) => {
       const { item } = click
       const hotbarSlot = this.hotbarStart + click.mouseButton
       const itemAtHotbarSlot = this.slots[hotbarSlot]
+      if (item === itemAtHotbarSlot) return
```

Comment:
is comparing by reference appropriate here?

Reply:
as long as references are passed, which currently is the case.
also `Item.equal` doesn't compare slot index, but that's important here

i also added a test case for this, so it's ok to compare by reference because the test passes

Reply:
so i changed it anyways so it doesn't break in the future :)